packages/commands/src/solana/deposit.ts

```diff
@@ -0,0 +1,210 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Wallet } from "@coral-xyz/anchor";
+import {
+  AccountLayout,
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+} from "@solana/spl-token";
+import { clusterApiUrl, PublicKey } from "@solana/web3.js";
+import GATEWAY_DEV_IDL from "@zetachain/protocol-contracts-solana/dev/idl/gateway.json";
+import GATEWAY_PROD_IDL from "@zetachain/protocol-contracts-solana/prod/idl/gateway.json";
+import * as bip39 from "bip39";
+import bs58 from "bs58";
+import { Command, Option } from "commander";
+import { ethers } from "ethers";
+import { z } from "zod";
+
+import {
+  SOLANA_NETWORKS,
+  SOLANA_TOKEN_PROGRAM,
+} from "../../../../types/shared.constants";
+import { handleError, validateAndParseSchema } from "../../../../utils";
+import { solanaDepositOptionsSchema } from "../../../../utils/solana.commands.helpers";
+
+type DepositOptions = z.infer<typeof solanaDepositOptionsSchema>;
+
+export const keypairFromMnemonic = async (
+  mnemonic: string
+): Promise<anchor.web3.Keypair> => {
+  const seed = await bip39.mnemonicToSeed(mnemonic);
+  const seedSlice = new Uint8Array(seed).slice(0, 32);
+  return anchor.web3.Keypair.fromSeed(seedSlice);
+};
+
+export const keypairFromPrivateKey = (
+  privateKey: string
+): anchor.web3.Keypair => {
+  try {
+    const decodedKey = bs58.decode(privateKey);
+    return anchor.web3.Keypair.fromSecretKey(decodedKey);
+  } catch (error) {
+    throw new Error(
+      "Invalid private key format. Expected base58-encoded private key."
+    );
+  }
+};
```

Comment:
These seem useful, we should probably extract them to a shared folder so they can be reused.